### ISSUE TITLE
Add support for adaptive inputstream addon to increase playable content from YoutubeDL

### ISSUE THREAD

**Issue by Tolriq** [Tolriq/repository.yatse.kodi#13]
As per title :)

Throwing .mpd file to Kodi is not enough :( it requires streaming an strm file with proper metadata at start ....

**Comment by iamsix** [comment]
From within the addon instead of a .strm you can use
```python
    playitem = xbmcgui.ListItem(path=entry['manifest_url'])
    playitem.setProperty('inputstreamaddon', 'inputstream.adaptive')
    playitem.setProperty('inputstream.adaptive.manifest_type', 'mpd')
    xbmc.Player().play(item=entry['manifest_url'], listitem=playitem)
```

Detecting when there's a DASH manifest (possibly preferring DASH over others?), and playing it within the addon with the playlist etc is another matter entirely - I just quickly threw that code in to share.py (replacing the 'm3u8' section) as a test and it does seem to work.

**Comment by Tolriq** [comment]
@iamsix Thanks that helps a lot :)

I've updated the source code here (but not push a new version yet). That handle mpd urls if inputstream addon is active.
It's tested with reddit and correctly use the dash playlist.

Would be nice if you can try and confirm it works, don't know what other site returns dash playlists.

**Comment by Tolriq** [comment]
Released in version 1.6.0 as a basic version, would require more work to support more cases and dash streams in playlists but require proper urls to get youtubedl data.
